src/react/deckgl.js

```diff
@@ -91,6 +91,17 @@ export default class DeckGL extends React.Component {
     }
     this._updateLayers(this.props);
 
+    const hasEvent = this.props.onLayerClick !== noop || this.props.onLayerHover !== noop;
```

Comment:
To make these checks less intrusive when reading this function, I'd add a one line comment on L93 explaining what this does.  And remove the empty line on L96 to make this a tight sub-block of code.